websites/A/Aodesu/presence.ts

```diff
@@ -0,0 +1,122 @@
+const presence = new Presence({
+		clientId: "1101840416866844672",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+let iFrameVideo: boolean,
+	currentTime: number,
+	duration: number,
+	paused: boolean,
+	playback: boolean,
+	reproductor: string;
+
+presence.on(
+	"iFrameData",
+	(data: {
+		iframeVideo: {
+			duration: number;
+			iFrameVideo: boolean;
+			currTime: number;
+			paused: boolean;
+			reproductor: string;
+		};
+	}) => {
+		playback = data.iframeVideo.duration !== null ? true : false;
+		reproductor = data.iframeVideo.reproductor;
+		if (playback) {
+			({ iFrameVideo, paused, duration } = data.iframeVideo);
+			currentTime = data.iframeVideo.currTime;
+		}
+	}
+);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+		largeImageKey: "https://i.imgur.com/CyVxys5.jpg",
+	};
+
+	presenceData.startTimestamp = browsingTimestamp;
+
+	if (document.location.pathname === "/") {
+		presenceData.smallImageKey = "home";
+		presenceData.smallImageText = "Página de inicio";
+		presenceData.details = "Viendo la página de inicio";
+	}else if(document.location.pathname === "/emitiendo"){
+		presenceData.smallImageKey = "emitting";
+		presenceData.smallImageText = "En Emisión";
+		presenceData.details = "Explorando animes en emisión";
+	}else if(document.location.pathname.startsWith("/ajustes")){
+		if(document.location.pathname === "/ajustes"){
+			presenceData.smallImageKey = "settings";
+			presenceData.smallImageText = "Ajustes";
+			presenceData.details = "Personalizando uwu"
+		}else if(document.location.pathname === "/ajustes/privacidad"){
+			presenceData.details = "Informandose 🧐"
+			presenceData.state = "Politicas de privacidad"
+		}
+	}else if(document.location.pathname.startsWith("/auth/")){
+		if(document.location.pathname === "/auth/iniciar-sesion"){
+			presenceData.smallImageKey = "login";
+			presenceData.smallImageText = "Inicio de sesión";
+			presenceData.details = "Iniciando sesión";
+		}else if(document.location.pathname === "/auth/registrarse"){
+			presenceData.smallImageKey = "register";
+			presenceData.smallImageText = "Registro de usuario";
+			presenceData.details = "¡Creando nueva cuenta!";
+		}
+	}else if(document.location.pathname.startsWith("/serie/")){
+		let title = document.querySelector("h1");
+		let season = document.querySelector("#season_id");
+		presenceData.smallImageKey = "preview";
+		presenceData.smallImageText= "Serie";
+		presenceData.details = `A punto de ver ${title.textContent}`
+		presenceData.state = `Temporada ${season.textContent}`
+		presenceData.buttons = [
+			{
+					label: "¡Ver También!",
+					url: `https://aodesu.com${document.location.pathname}`
```

Comment:
For easier maintainability, use the `document.location.href` attribute here.
```suggestion
					url: href,
```